public/app/features/explore/ContentOutline/ContentOutline.tsx

```diff
@@ -1,46 +1,37 @@
-import { css } from '@emotion/css';
+import { css, cx } from '@emotion/css';
 import React, { useEffect, useRef, useState } from 'react';
 import { useToggle, useScroll } from 'react-use';
 
 import { GrafanaTheme2 } from '@grafana/data';
 import { reportInteraction } from '@grafana/runtime';
 import { useStyles2, PanelContainer, CustomScrollbar } from '@grafana/ui';
 
-import { useContentOutlineContext } from './ContentOutlineContext';
+import { ContentOutlineItemContextProps, useContentOutlineContext } from './ContentOutlineContext';
 import { ContentOutlineItemButton } from './ContentOutlineItemButton';
 
-const getStyles = (theme: GrafanaTheme2) => {
-  return {
-    wrapper: css({
-      label: 'wrapper',
-      position: 'relative',
-      display: 'flex',
-      justifyContent: 'center',
-      marginRight: theme.spacing(1),
-      height: '100%',
-      backgroundColor: theme.colors.background.primary,
-    }),
-    content: css({
-      label: 'content',
-      top: 0,
-    }),
-    buttonStyles: css({
-      textAlign: 'left',
-      width: '100%',
-      padding: theme.spacing(0, 1.5),
-    }),
-  };
+const INDENT_LEVELS = {
+  ROOT: '32px',
+  CHILD_COLLAPSED: '32px',
+  CHILD_EXPANDED: '66px',
 };
 
 export function ContentOutline({ scroller, panelId }: { scroller: HTMLElement | undefined; panelId: string }) {
-  const { outlineItems } = useContentOutlineContext();
-  const [expanded, toggleExpanded] = useToggle(false);
-  const [activeItemId, setActiveItemId] = useState<string | undefined>(outlineItems[0]?.id);
-  const styles = useStyles2((theme) => getStyles(theme));
+  const [contentOutlineExpanded, toggleContentOutlineExpanded] = useToggle(false);
+  const styles = useStyles2((theme) => getStyles(theme, contentOutlineExpanded));
+  const [sectionExpanded, toggleSectionExpanded] = useToggle(false);
   const scrollerRef = useRef(scroller || null);
   const { y: verticalScroll } = useScroll(scrollerRef);
+  const { outlineItems } = useContentOutlineContext() ?? { outlineItems: [] };
+  const [activeSectionId, setActiveSectionId] = useState<string | undefined>(outlineItems[0]?.id);
+  const [activeSectionChildId, setActiveSectionChildId] = useState<string | undefined>(
```

Comment:
the type param shouldn't be needed here
```suggestion
  const [activeSectionChildId, setActiveSectionChildId] = useState(
```